subsys/bluetooth/services/dis.c

```diff
@@ -38,6 +47,33 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#ifdef CONFIG_BT_GATT_DIS_DYNAMIC
+#define DIS_STR_MAX CONFIG_BT_GATT_DIS_STR_MAX
+#define DIS_STORAGE /* empty */
+#else
+#define DIS_STR_MAX /* empty */
+#define DIS_STORAGE const
+#endif
+
+static DIS_STORAGE u8_t dis_model[DIS_STR_MAX] = CONFIG_BT_GATT_DIS_MODEL;
```

Comment:
I don't get why we need to declare a variable to store the config options? It seems to be we would be wasting memory as one would have to assign CONFIG_BT_GATT_DIS_STR_MAX with the biggest string size.

Reply:
This is only the case when reading from the settings system. Otherwise it is just a const with length defined by the kconfig.

Reply:
Can see the the const concept poses some problems during checking, I will work out something else.

Reply:
Now using kconfig directly when not using settings system